tests/test_indexes.py

```diff
@@ -21,6 +23,11 @@
     check,
 )
 
+if TYPE_CHECKING:
+    from pandas.core.indexes.timedeltas import TimedeltaIndex
+else:
+    TimedeltaIndex: TypeAlias = pd.Index
+
```

Comment:
I don't see why the `if` is needed.  Why can't you just import `TimedeltaIndex` ?